### DISCUSSION TITLE
Alpha Rule

### DISCUSSION THREAD

**Discussion by wizzymore** [adonisjs/core#2573] [Ideas]
Hi guys,

Currently i can see that the rules.alpha currently doesn't support "dots"
Is there any other way to allow only dots without regex, and if not, couldn't this be added to alpha?

**Comment by thetutlage** [comment]
Is it common for alpha string to have `dots`?

**Reply by wizzymore** [reply]
No, but neither is it common to contain the other stuff that it currently allows.
Normally it should be only `a-z` `A-Z` `0-9`, but it would be could to have an easier check for usernames without doing regex manually as there are already the three restrictions to the rule.

```javascript


{
  username: schema.string({}, [
    rules.alpha({
      allow: ['space', 'underscore', 'dash']
    })
  ])
}
```

**Reply by thetutlage** [reply]
Yup, It is fine to allow it via options. I thought you want `dot` to be allowed all the times. Feel free to send a PR for same
